assets/src/components/AssignmentTable.js

```diff
@@ -156,6 +179,19 @@ function AssignmentTable (props) {
     setAssignmentGroupNames(uniqueGroupNames)
   }, [assignments])
 
+  useEffect(() => {
+    setFiltersAreClear(assignmentNameFilter.trim().length === 0 && assignmentStatusFilterArray.length === 0 && assignmentGroupFilterArray.length === 0)
+  }, [assignmentNameFilter, assignmentStatusFilterArray, assignmentGroupFilterArray])
+
+  useEffect(() => {
+    setFilteredAssignments(
+      assignments
+        .filter(assignment => assignmentNameFilter.trim().length === 0 || assignment.name.toUpperCase().includes(assignmentNameFilter.toUpperCase()))
+        .filter(assignment => assignmentStatusFilterArray.length === 0 || (assignmentStatusFilterArray.indexOf(assignmentStatus.GRADED) >= 0 && assignment.graded) || (assignmentStatusFilterArray.indexOf(assignmentStatus.SUBMITTED) >= 0 && assignment.submitted && !assignment.graded) || (assignmentStatusFilterArray.indexOf(assignmentStatus.UNSUBMITTED) >= 0 && !(assignment.graded || assignment.submitted)))
+        .filter(assignment => assignmentGroupFilterArray.length === 0 || assignmentGroupFilterArray.indexOf(assignment.assignmentGroup.name) >= 0)
```

Comment:
These 3 filter blocks inner logic can be separated as `const` name reflecting what it is doing and simply applied

This is just a pseudo code
```
const filterByAssignmentName = () => {
return assignment => assignmentNameFilter.trim().length === 0 || assignment.name.toUpperCase().includes(assignmentNameFilter.toUpperCase())
}
```

I think you can also separate the functions to utils folder, if you prefer as I see that the `AssignmentTable.js` is becoming very huge

Reply:
Refactored the filter logic into individual functions